R/is.R

```diff
@@ -28,244 +28,14 @@ is_auto <- function(arg) {
-    deparse(call$arg),
-    " = ",
-    eval(call$arg, envir = env),
-    " is not a valid time entry.\n",
-    "time_imputation should be specified as 'hh:mm:ss' (e.g. '00:00:00') or ",
-    "'FIRST', 'LAST' to get the first/last time of the day"
-  )
-}
-
-#' Check Validity of the Minute/Second Portion of the Time Input
-#'
-#' Minutes and seconds are expected to range from 0 to 59
-#'
-#' @param arg The argument to check
-#'
-#' @author Samia Kabi
-#'
-#' @return `TRUE` if the argument is a valid min/sec input, `FALSE` otherwise
-#'
-#' @keywords is
-#' @family is
-#' @export
-is_valid_sec_min <- function(arg) {
-  arg %in% 0:59
-}
-on_failure(is_valid_sec_min) <- function(call, env) {
-  paste0(
-    "Argument ",
-    deparse(call$arg),
-    " = ",
-    eval(call$arg, envir = env),
-    " is not a valid min/sec.\n",
-    "Values must be between between 0-59"
-  )
-}
-
-#' Check Validity of the Hour Portion in the Time Input
-#'
-#' Hours are expected to range from 0 to 23
-#'
-#' @param arg The argument to check
-#'
-#' @author Samia Kabi
-#'
-#' @return `TRUE` if the argument is a valid hour input, `FALSE` otherwise
-#'
-#' @keywords is
-#' @family is
-#' @export
-is_valid_hour <- function(arg) {
-  arg %in% 0:23
-}
-on_failure(is_valid_hour) <- function(call, env) {
-  paste0(
-    "Argument ",
-    deparse(call$arg),
-    "=",
-    eval(call$arg, envir = env),
-    " is not a valid hour.\n",
-    "Values must be between 0-23"
-  )
-}
-
-#' Check Validity of the Day Portion in the Date Input
-#'
-#' Days are expected to range from 1 to 31
-#'
-#' @param arg The argument to check
-#'
-#' @author Samia Kabi
-#'
-#' @return `TRUE` if the argument is a day input, `FALSE` otherwise
-#'
-#' @keywords is
-#' @family is
-#' @export
-is_valid_day <- function(arg) {
-  arg %in% 1:31
-}
-on_failure(is_valid_day) <- function(call, env) {
-  paste0(
-    "Argument ",
-    deparse(call$arg),
-    " = ",
-    eval(call$arg, envir = env),
-    " is not a valid day.\n",
-    "Values must be between 1-31"
-  )
-}
-
-#' Check Validity of the Month Portion in the Date Input
-#'
-#' Days are expected to range from 1 to 12
-#'
-#' @param arg The argument to check
-#'
-#' @author Samia Kabi
-#'
-#' @return `TRUE` if the argument is a month input, `FALSE` otherwise
-#'
-#' @keywords is
-#' @family is
-#'
-#' @export
-is_valid_month <- function(arg) {
-  arg %in% 1:12
-}
-on_failure(is_valid_month) <- function(call, env) {
-  paste0(
-    "Argument ",
-    deparse(call$arg),
-    " = ",
-    eval(call$arg, envir = env),
-    " is not a valid month.\n",
-    "Values for month must be between 1-12. ",
-    "Please check the date_imputation input: it should be sepcified as 'dd-mm'"
-  )
-}
-
 #' Is order vars?
 #'
+#' Check if inputs are created using `vars()` or calls involving `desc()`
 #' @param arg A<n R object
 #'
-#' @return A `logical` vector
+#' @return A TRUE if input was created with `vars()` or calls involving `desc()`
+#' otherwise throws an error
```

Comment:
The function does not throw an error but return `FALSE` if the argument is not a list of order vars.